front/app/hooks/useTopics.ts

```diff
@@ -1,51 +1,30 @@
 import { useState, useEffect } from 'react';
-import { projectTopicsStream } from 'services/projectTopics';
 import {
   ITopicData,
   topicByIdStream,
   topicsStream,
   Code,
 } from 'services/topics';
 import { Observable, of, combineLatest } from 'rxjs';
-import { map, switchMap } from 'rxjs/operators';
-import { isNilOrError } from 'utils/helperUtils';
+import { map } from 'rxjs/operators';
+import { isNilOrError, NilOrError, reduceErrors } from 'utils/helperUtils';
 
 interface Parameters {
-  projectId?: string;
   topicIds?: string[];
   code?: Code;
   exclude_code?: Code;
   sort?: 'new' | 'custom';
 }
 
 export default function useTopics(parameters: Parameters) {
-  const { projectId, topicIds, code, exclude_code, sort } = parameters;
-  const [topics, setTopics] = useState<
-    (ITopicData | Error)[] | undefined | null | Error
-  >(undefined);
+  const { topicIds, code, exclude_code, sort } = parameters;
+  const [topics, setTopics] = useState<ITopicData[] | NilOrError>(undefined);
   const queryParameters = { code, exclude_code, sort };
 
   useEffect(() => {
     let observable: Observable<(ITopicData | Error)[] | null> = of(null);
 
```

Comment:
I have removed the possibility of using `projectId`, because it simplifies the code and because it would be ambiguous which topics you mean (the 'project topics' or the 'project allowed input topics'). Instead how it works now is you first get the correct topic IDs by using `useProjectAllowedInputTopics`, and then call the `useTopics` hook with those `topicIds`.